tests/converter_test.php

```diff
@@ -465,4 +465,80 @@ public function test_client_uses_proxy() {
+        $generator = $this->getDataGenerator()->get_plugin_generator('mod_assign');
+        $instance = $generator->create_instance(array('course' => $course->id));
+        $context = context_module::instance($instance->cmid);
+
+        // Create file to analyze.
+        $fs = get_file_storage();
+        $filerecord = array(
+            'contextid' => $context->id,
+            'component' => 'assignsubmission_file',
+            'filearea' => 'submission_files',
+            'itemid' => $instance->cmid,
+            'filepath' => '/',
+            'filename' => 'testsubmission.odt');
+        $fileurl = $CFG->dirroot . '/files/converter/librelambda/tests/fixtures/testsubmission.odt';
+        $file = $fs->create_file_from_pathname($filerecord, $fileurl);
+
+        $conversiondata = (object) [
+            'sourcefileid' => $file->get_id(),
+            'targetformat' => 'pdf',
+            'converter' => '\fileconverter_librelambda\converter',
+        ];
+
+        // First create a valid conversion record.
+        $conversion = new conversion(0, $conversiondata);
+        $conversion->create();
+
+        // Now lets duplicate this a couple of times
+        $conversion2 = new conversion(0, $conversiondata);
+        $conversion2->create();
+
+        $conversion3 = new conversion(0, $conversiondata);
+        $conversion3->create();
+
+        // Setup AWS mocking.
+        $mock = new MockHandler();
+        $mock->append(new Result(array('ObjectURL' => 's3://herpderp')));
+        $mock->append(function (CommandInterface $cmd, RequestInterface $req) {
+            return new S3Exception('Mock exception', $cmd, array('code' => 'NoSuchKey'));
+        });
+
+        $converter = new \fileconverter_librelambda\converter();
+        $converter->create_client($mock);
+        $converter->start_document_conversion($conversion);
+        $converter->start_document_conversion($conversion2);
+        $converter->start_document_conversion($conversion3);
+
+        // Now check the status of the table before and after executing the task.
+        $this->assertEquals(3, $DB->count_records('file_conversion'));
+        $task = new \fileconverter_librelambda\task\convert_submissions();
+        $this->expectOutputRegex("/Processing/"); // We expect trace output for this test.
+        $task->execute();
+        $this->assertEquals(1, $DB->count_records('file_conversion'));
+
+        // Now check that we aren't being greedy and deleting non librelambda records.
+        $conversiondata->converter = '\fileconverter_googledrive\converter';
+        $conversion4 = new conversion(0, $conversiondata);
+        $conversion4->create();
+        $converter->start_document_conversion($conversion4);
+
+        // Now check the status of the table before and after executing the task.
+        $this->assertEquals(2, $DB->count_records('file_conversion'));
+        $task->execute();
+        $this->assertEquals(2, $DB->count_records('file_conversion'));
```

Comment:
This test is not sufficient in determining if the googledrive converter remains after the execute call.

An example of where this would fail is if there were only 2 conversions present `$conversion, $conversion4` and it was executed. It would remove the googledrive converter in the array_shift (weird I know) because get_conversions_for_file seems to return (at least for me) in descending order. So google drive converter is first in the array and is shifted off / removed.